### PR TITLE
Permissionless deployment

### PR DESCRIPTION
**Pull request by asaj**
### Description

_What's included in this PR?_

### Drive-by changes

_Are there any minor or drive-by changes also included?_

### Related issues

- Fixes #[issue number here]

### Backward compatibility

_Are these changes backward compatible?_

Yes
No

_Are there any infrastructure implications, e.g. changes that would prohibit deploying older commits using this infra tooling?_

None
Yes

### Testing

_What kind of testing have these changes undergone?_

None
Manual
Unit Tests

### CODE REVIEW

File: solidity/scripts/DeployCore.s.sol [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+
+import {Mailbox} from "../contracts/Mailbox.sol";
+import {InterchainGasPaymaster} from "../contracts/InterchainGasPaymaster.sol";
+import {ProxyAdmin} from "../contracts/upgrade/ProxyAdmin.sol";
+import {TransparentUpgradeableProxy} from "../contracts/upgrade/TransparentUpgradeableProxy.sol";
+import {MultisigIsm} from "../contracts/isms/MultisigIsm.sol";
+import {DeployLib} from "./lib/DeployLib.sol";
+
+contract DeployCore is Script {
+    function deployIgp(address proxyAdmin)
```
Comment (yorhodes) [inline solidity/scripts/DeployCore.s.sol:16]: you could try `using DeployLib for Vm`
and then `vm.getDomainId(local)`

File: solidity/scripts/DeployCore.s.sol [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+    function run() public {
+        // Read all the config we need first so that we ensure valid
+        // config before sending any transactions.
+        address owner = vm.envAddress("OWNER");
+        uint256 deployerPrivateKey = vm.envUint("PRIVATE_KEY");
+        string memory local = vm.envString("LOCAL");
+        uint32 localDomain = DeployLib.getDomainId(vm, local);
+        string[] memory remotes = vm.envString("REMOTES", ",");
+        DeployLib.MultisigIsmConfig[] memory configs = DeployLib
+            .getMultisigIsmConfigs(vm, remotes);
```
Comment (yorhodes) [inline solidity/scripts/DeployCore.s.sol:61]: why mix env with JSON config?

File: solidity/scripts/DeployCore.s.sol [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+        TransparentUpgradeableProxy proxy = new TransparentUpgradeableProxy(
+            address(igp),
+            proxyAdmin,
+            initData
+        );
+        return InterchainGasPaymaster(address(proxy));
+    }
+
+    function deployMailbox(
+        address deployer,
```
Comment (yorhodes) [inline solidity/scripts/DeployCore.s.sol:34]:
```suggestion
        address owner,
```

File: solidity/scripts/DeployCore.s.sol (hunk 1 above)
Comment (yorhodes) [inline solidity/scripts/DeployCore.s.sol:16]:
```suggestion
    function deployIgp(ProxyAdmin proxyAdmin)
```

File: solidity/scripts/DeployCore.s.sol [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+        address proxyAdmin,
```
Comment (yorhodes) [inline solidity/scripts/DeployCore.s.sol:35]:
```suggestion
        ProxyAdmin proxyAdmin,
```

**Review by yorhodes — COMMENTED** [review]
glad we could reuse some of my rabbit hole :)

**Comment by asaj** [comment]
Closing in favor of the hyperlane-deploy repo

**Pull request closed without merging**